### PR TITLE
kvserver: remove unused snapshot log entry code

### PR DESCRIPTION
**Pull request by tbg**
We used to have to send the raft log along with snapshots as a result of
(in the olden days) requiring all replicas to agree on the truncated
state. This hasn't been (generally) true as of v19.1 (#35701), though it
was still a possibility until v22.1 (#58088). This commit removes the
corresponding code.

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/store_snapshot.go [hunk 1]
```diff
@@ -686,7 +572,7 @@ func (s *Store) receiveSnapshot(
 
 	// Defensive check that any snapshot contains this store in the	descriptor.
 	storeID := s.StoreID()
-	if _, ok := header.State.Desc.GetReplicaDescriptor(storeID); !ok {
+	if _, ok := header.State.Desc.GetReplicaDescriptor(storeID); !ok { // TODO
```
Comment (erikgrinaker) [inline pkg/kv/kvserver/store_snapshot.go:575]: `TODO` what?

**Review by erikgrinaker — APPROVED** [review]
🧹

File: pkg/kv/kvserver/store_snapshot.go (hunk 1 above)
Comment (tbg) [reply pkg/kv/kvserver/store_snapshot.go]: Oops, this slipped in. Removed.

**Comment by tbg** [comment]
Just fixed the lint failures and `kvserver` passed locally, so going to try

bors r=erikgrinaker
TFTR!

**Pull request merged**